tests/src/Unit/RequestSubscriberTest.php

```diff
@@ -0,0 +1,347 @@
+      $route_match,
+      $this->logger,
+      $this->time
+    );
+    $this->assertArrayEquals($agent_options, $this->invokeMethod(
+      $request_subscriber,
+      'prepareAgentOptions',
+      []
+    ));
+  }
+
+  /**
+   * Tests the constructQuerySpan method.
+   *
+   * ::covers constructQuerySpan.
+   */
+  public function testConstructQuerySpan() {
+    // Test constructQuerySpan.
+    // Create necessary classes to pass to the RequestSubscriber constructor.
+    $agent_options = [];
+    $agent = $this->prophesize(Agent::class);
+    $agent = $agent->reveal();
+    $api_service = $this->prophesize(ApiService::class);
+    $api_service->isEnabled()->willReturn(TRUE);
+    $api_service->isConfigured()->willReturn(TRUE);
+    $api_service->getAgent($agent_options)->willReturn($agent);
+    $api_service = $api_service->reveal();
+
+    $options = [];
+    $route = $this->prophesize(Route::class);
+    $route->getOptions()->willReturn($options);
+    $route = $route->reveal();
+    $route_match = $this->prophesize(RouteMatchInterface::class);
+    $route_match->getRouteObject()->willReturn($route);
+    $route_match = $route_match->reveal();
+
+    $request_subscriber = new RequestSubscriber(
+      $api_service,
+      $route_match,
+      $this->logger,
+      $this->time
+    );
+    $connection_name = 'testConnectionName';
+    $driver_name = 'testDriverName';
+    $expected_result = $this->getExpectedSpanData(
+      $connection_name,
+      $driver_name,
+      $this->getQueryForSpan()
+    );
+    $actual_result = $this->invokeMethod(
+      $request_subscriber,
+      'constructQuerySpan',
+      [
+        $connection_name,
+        $driver_name,
+        $this->getQueryForSpan(),
+      ]
+    );
+    $this->assertArrayEquals($expected_result, $actual_result);
+  }
+
+  /**
+   * Tests the constructDatabaseSpans method.
+   *
+   * ::covers constructDatabaseSpans.
+   */
+  public function testConstructDatabaseSpans() {
+    // Test constructDatabaseSpans.
+    // Create necessary classes to pass to the RequestSubscriber constructor.
+    $agent_options = [];
+    $agent = $this->prophesize(Agent::class);
+    $agent = $agent->reveal();
+    $api_service = $this->prophesize(ApiService::class);
+    $api_service->isEnabled()->willReturn(TRUE);
+    $api_service->isConfigured()->willReturn(TRUE);
+    $api_service->getAgent($agent_options)->willReturn($agent);
+    $api_service = $api_service->reveal();
+
+    $options = [];
+    $route = $this->prophesize(Route::class);
+    $route->getOptions()->willReturn($options);
+    $route = $route->reveal();
+    $route_match = $this->prophesize(RouteMatchInterface::class);
+    $route_match->getRouteObject()->willReturn($route);
+    $route_match = $route_match->reveal();
+
+    $request_subscriber = new RequestSubscriber(
+      $api_service,
+      $route_match,
+      $this->logger,
+      $this->time
+    );
+
+    $this->assertArrayEquals([], $this->invokeMethod(
```

Comment:
How can we mock the ```Database::getAllConnectionInfo()``` and ```$database_log = Database::startLog('elastic_apm', $key);``` calls so that we can test this better with different options.